Centiq/sap-hana
# BoM Media Copying

## Problem

At present installation media has to be copied onto target SCS manually.

## Solution

This PR adds a role which recursively travels BoMs and copies media from the specified Azure Storage Account to the SCS.

## Tests / Acceptance Criteria

1. From your deployer (aka rti), as the `azureadm` user in the `sap-hana` directory:

   ```text
   cd deploy/ansible
   ```

1. Edit `poc_bom_iteration_playbook.yml` and ensure that appropriate values are used for:

   - `sapbits_location_base_path`
   - `bom_base_name`
   - Ensure `test_mode` is `true`.

   For example:

   ```text
   ---

   - hosts: localhost
     vars:
       sapbits_location_base_path: "https://npeus2saplibb546.blob.core.windows.net/sapbits"
       bom_base_name: "S4HANA_2020_ISS_v001"
       test_mode: true

     roles:

       - role: process_bom
   ```

1. Execute the demo playbook:

   ```text
   ansible-playbook -i ~/hosts poc_bom_iteration_playbook.yml
   ```

   - [ ] All expected media files are reported with expected copy locations(s) as per the BoM(s).
   - [ ] Code LGTM

## Todo

Following approval:

- [ ] Remove "test_mode" functionality. The main code flow has diverged too far and could be better addressed with proper tests.
- [ ] Remove `tasks/configure_disk.yml`
- [ ] Remove `poc_bom_iteration_playbook.yml`
